src/main/java/io/javalin/Javalin.java

```diff
@@ -732,4 +734,34 @@ void clearMatcherAndMappers() {
         exceptionMapper.getExceptionMap().clear();
     }
 
+    // Extension
+
+    /**
+     * Registers an {@link Extension} with the Javalin application.
+     *
+     * @param extension You're free to implement the extension as a class or a lambda expression.
+     * @return Self instance for fluent, method-chaining API
+     */
+    public Javalin extension(Extension extension) {
+        extension.register(this);
+
+        String clazzName = extension.getClass().getCanonicalName();
```

Comment:
I guess you can use the class object as a key directly, what do you think?

Reply:
That would store lambdas with their generated class name in the map. I thought to not store anonymous classes and lambdas in the map.

See https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#getCanonicalName--